pkg/services/ngalert/state/state.go

```diff
@@ -28,6 +28,28 @@ type Evaluation struct {
 	EvaluationTime   time.Time
 	EvaluationState  eval.State
 	EvaluationString string
+	// Values contains the RefID and value of reduce and math expressions.
+	// It does not contain values for classic conditions as the values
+	// in classic conditions do not have a RefID.
+	Values map[string]EvaluationValue
+}
+
+// EvaluationValue contains the labels and value for a RefID in an evaluation.
+type EvaluationValue struct {
+	Labels data.Labels
+	Value  *float64
+}
+
+// NewEvaluationValues returns the labels and values for each RefID in the capture.
+func NewEvaluationValues(m map[string]eval.NumberValueCapture) map[string]EvaluationValue {
```

Comment:
Apologies if this is a dumb question, but why do we need this conversion? Couldn't we just use `map[string]eval.NumberValueCapture` directly? We're already converting these in the template to `templateCaptureValue`.

Reply:
I wanted to create a separation between the evaluation results and what we store in the cache as I think it will make it easier to make isolated changes to the code in future. It's not something I feel that strong about though and I'm more than open to changing it.

Reply:
If the only purpose is the separation of concerns - I'd be in favour of removing it, it's not doing much for us and the only way this is exposed to users is through the template, so we're covered in that area.

Reply:
@gotjosh I think it might be good to have a different type here. Since we expose Go templates, I think it is important to have a very well defined interface. Else it can become possible for properties to get added to types, and this leaks into what can be accessed in a template.

Reply:
I 100% agree with that, but I thought that was:

```golang
// templateCaptureValue represents each value in .Values in the annotations
// and labels template.
type templateCaptureValue struct {
	Labels map[string]string
	Value  *float64
}
```

and not `EvaluationValue`, it seems like we're doing that twice if I'm reading this right.

Reply:
Had an offline discussion with @gerobinson and I'm happy to leave this as is.